### PR TITLE
Factorize PythonParameterSet code from other fwk to reduce dependencies

### PR DESCRIPTION
**Pull request by davidlange6**
Factorize PythonParameterSet code from other fwk to reduce dependencies and make it easier to replace with a different one. Somewhat a work in progress but getting big enough to get comments on and perhaps merge.

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: FWCore/ParameterSetReader/interface/ParameterSetReader.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+namespace edm {
+
+  std::unique_ptr<edm::ParameterSet> getPSetFromConfig(const std::string &config);
+
+  std::unique_ptr<edm::ParameterSet> readConfig(std::string const& config, int argc, char* argv[]);
+
+  std::unique_ptr<ParameterSet> readConfig(std::string const& config);
```
Comment (fwyzard) [inline FWCore/ParameterSetReader/interface/ParameterSetReader.h]: cn you add `edm::` also here for construct ?

File: FWCore/Framework/interface/EventProcessor.h [hunk 2]
```diff
@@ -79,15 +79,15 @@ namespace edm {
     // the specification of services by name only (defaultServices and forcedServices).
     // 'defaultServices' are overridden by 'config'.
     // 'forcedServices' the 'config'.
-    explicit EventProcessor(std::string const& config,
+    explicit EventProcessor(std::unique_ptr<ParameterSet> parameterSet,//std::string const& config,
```
Comment (fwyzard) [inline FWCore/Framework/interface/EventProcessor.h]: can you drop the comment ?

**Comment by fwyzard** [comment]
I am curious: what is the goal here?
To eventually implement a mechanism that reads the python configuration without relying on Boost::Python ?

**Comment by davidlange6** [comment]
For example (perhaps not so eventually..) - its come out in looking at easy ways to progress towards python3. As it stands, boost-python is pretty deep in the code stack

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

File: FWCore/Framework/interface/EventProcessor.h [hunk 3]
```diff
@@ -96,7 +96,7 @@ namespace edm {
                    serviceregistry::ServiceLegacy legacy);
 
     /// meant for unit tests
-    EventProcessor(std::string const& config, bool isPython);
+    EventProcessor(std::unique_ptr<ParameterSet> parameterSet, bool isPython);//std::string const& config, bool isPython);
```
[the remark this answers is not in the document]
Comment (davidlange6) [reply FWCore/Framework/interface/EventProcessor.h]: Right. the isPython argument saves this constructor from conflicting with another one.. lets see if I can unify them

**Comment by davidlange6** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/24620	d9a193b 
- changes in reco packages are technical (header or method renaming)
- jenkins tests pass and comparisons with the baseline show no relevant differences (ignoring random diffs in L1 and muon DQM)

**Comment by davidlange6** [comment]
@dr15jones - did you have more comments on this PR?

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
@ggovi @thomreis @rekovic @lpernie @tocheng @pohsun could you please check this PR and sign it in case?

**Comment by thomreis** [comment]
+1

File: CondFormats/CSCObjects/test/testCSCMapping.cpp [hunk 4]
```diff
@@ -71,7 +71,8 @@ int testCSCMapping::runIt(const std::string& config){
   edm::AssertHandler ah;
   int rc=0;
   try {
-    edm::EventProcessor proc(config);
+    std::unique_ptr<edm::ParameterSet> pset; // ???????
```
Comment (fabiocos) [inline CondFormats/CSCObjects/test/testCSCMapping.cpp]: @davidlange6 sorry why pset and not edm::getPSetFromConfig(config) as in the next file?

**Comment by ggovi** [comment]
+1

File: CondFormats/CSCObjects/test/testCSCMapping.cpp (hunk 4 above)
Comment (davidlange6) [reply CondFormats/CSCObjects/test/testCSCMapping.cpp]: ah - thats a bug I think -looking

**Comment by davidlange6** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

File: CondFormats/CSCObjects/test/testCSCMapping.cpp [hunk 5]
```diff
@@ -71,7 +71,7 @@ int testCSCMapping::runIt(const std::string& config){
   edm::AssertHandler ah;
   int rc=0;
   try {
-    edm::EventProcessor proc(config);
+    edm::EventProcessor proc(edm::getPSetFromConfig(config));
```
Comment (fabiocos) [inline CondFormats/CSCObjects/test/testCSCMapping.cpp]: @davidlange6 you miss 

#include "FWCore/ParameterSetReader/interface/ParameterSetReader.h"

Not sure why before it was working with pset

**Comment by slava77** [comment]
@davidlange6 

please clarify on the status of this PR (it looks like it needs an update to avoid the build errors).
Thank you.

File: CondFormats/CSCObjects/test/testCSCMapping.cpp (hunk 5 above)
Comment (davidlange6) [reply CondFormats/CSCObjects/test/testCSCMapping.cpp]: Right - it was working before as nothing from ParameterSetReader class is used to define the PSet class structure itself.

**Comment by davidlange6** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/24620	b9adb4c
- updates since the last signoff by reco are somewhat technical and do not affect reco code https://github.com/cms-sw/cmssw/compare/d9a193b37996dc3ee67bc921101839e2a2a7f5e1...b9adb4c98882864402ed20ed8f24aac508e64465
- jenkins tests pass

**Comment by fabiocos** [comment]
@davidlange6 this PR was already signed by everybody, so ok for me to move forward without waiting a full second round of signatures

**Comment by fabiocos** [comment]
actually alca looked missing, @lpernie @tocheng @pohsun could you please check and sign it in case?

**Comment by pohsun** [comment]
+1

**Comment by fabiocos** [comment]
+1

l1 and db had already signed the PR before the last change, that would be unrelated

**Comment by fabiocos** [comment]
please test

**Comment by slava77** [comment]
> Reco comparison results: 2468 differences found in the comparisons

and the commit was older than the test IB.
The changes are in the PUpmx 250202.181 and look like a change that started at digi level or earlier

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
+1

differences in DQM appear unrelated to this PR

**Comment by fabiocos** [comment]
merge

**Pull request merged**